### DISCUSSION TITLE
Using Laravel PHP v8 with modes-tcp-client for Solar Inverter remote control - possible?

### DISCUSSION THREAD

**Discussion by vapordev** [aldas/modbus-tcp-client#105] [Q&A]
Hi there,

We are in need of software that will work with Laravel PHP v8 to integrate with a solar inverter. We are in need of read/write information to and from the inverter like energy created, stored, used, battery levels, etc. Allow for remote setup with WIFI, charge and discharge battery functions from a Laravel panel (this Laravel panel we have already built) Currently there is a TypeScript setup to read between Modbus and PHP. Will this software be ideal for us and remove the TypeScript requirements?

**Comment by aldas** [comment]
You have to try it out.  See the examples https://github.com/aldas/modbus-tcp-client/tree/master/examples

Before you start - make sure you know:
* is your inverter using Modbus TCP or Modbus RTU or  Modbus RTU over TCP
* what you inverter unitID/slaveID. By default this is left to 0 in examples and you need to set if you need something else (3rd parameter)

**Reply by vapordev** [reply]
Hi, thank you for getting back to me. I can confirm the inverters use RTU

**Reply by aldas** [reply]
Reading (FC4) from serial device (usb in that example) can be found here https://github.com/aldas/modbus-tcp-client/blob/master/examples/rtu_usb_to_serial_stream.php

**Reply by vapordev** [reply]
ok, so slaveID is default 1.
This code is totally new to us and we do not understand it. Is there anyone you can recommend to help us on a job to create communications from Laravel php to Modbus? We are happy to agree a price.
